clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -584,27 +576,20 @@ private void handleConnections(long now) {
             log.debug("Completed connection to node {}", node);
             // Though the node is connected, we might not still be able to send requests. For instance,
             // in case of SSL connection, SSL handshake happens after connection is established.
-            if (this.requiredApiVersions == null) {
-                this.connectionStates.ready(node);
-            } else {
-                this.connectionStates.checkingApiVersions(node);
-                nodesNeedingApiVersionsFetch.add(node);
-            }
+            this.connectionStates.checkingApiVersions(node);
+            nodesNeedingApiVersionsFetch.add(node);
         }
-        maybeInitiateApiVersionsFetch(now);
     }
 
-    private void maybeInitiateApiVersionsFetch(long now) {
-        if (this.requiredApiVersions == null)
```

Comment:
That's a good point.  What do you suggest?  One approach is to (re)add a constructor parameter to NetworkClient that will allow the brokers to bypass sending ApiVersions.